com.avaloq.tools.ddk.xtext.generator/src/com/avaloq/tools/ddk/xtext/generator/parser/antlr/AcfKeywordHelper.java

```diff
@@ -0,0 +1,293 @@
+    Set<String> result = new TreeSet<>(KeywordHelper.keywordComparator);
+    for (CharSequence cs : keywordValueToToken.keySet()) {
+      result.add(cs.toString());
+    }
+
+    return Collections.unmodifiableSet(result);
+  }
+
+  /**
+   * Initializes the keyword map.
+   * Searches for keywords that belong to grammar rules reachable from the first rule of the given grammar.
+   *
+   * @param grammar
+   *          Root grammar
+   * @return Keyword map
+   */
+  private BiMap<CharSequence, String> createKeywordMap(final Grammar grammar) {
+    List<ParserRule> parserRules = ParserSemanticPredicatesUtil.allReachableParserRules(grammar);
+    List<EnumRule> enumRules = ParserSemanticPredicatesUtil.allReachableEnumRules(grammar);
+    Iterator<EObject> iter = Iterators.concat(EcoreUtil.<EObject> getAllContents(parserRules), EcoreUtil.<EObject> getAllContents(enumRules));
+    Iterator<Keyword> filtered = Iterators.filter(iter, Keyword.class);
+    Iterator<String> transformed = Iterators.transform(filtered, new Function<Keyword, String>() {
+      @Override
+      public String apply(final Keyword from) {
+        return from.getValue();
+      }
+    });
+    TreeSet<String> treeSet = Sets.newTreeSet(new Comparator<String>() {
+      @Override
+      public int compare(final String o1, final String o2) {
+        if (o1.length() == o2.length()) {
+          return o1.compareTo(o2);
+        }
+        return Integer.valueOf(o1.length()).compareTo(Integer.valueOf(o2.length()));
+      }
+    });
+    Iterators.addAll(treeSet, transformed);
+    BiMap<CharSequence, String> result = HashBiMap.create();
+    int i = 1;
+    for (String s : treeSet) {
+      CharSequence key = createKey(s);
+      if (!result.containsKey(key)) {
+        result.put(key, createKeywordName(i, s));
+        i++;
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Generates a name for keyword grammar rule.
+   *
+   * @param count
+   *          Numeric id for the name to ensure uniqueness
+   * @param value
+   *          Original value of the keyword
+   * @return Rule name
+   */
+  private String createKeywordName(final int count, final String value) {
+    StringBuilder name = new StringBuilder();
+    name.append("KEYWORD_"); //$NON-NLS-1$
+    name.append(count);
+    name.append('_');
+    for (char ch : value.toCharArray()) {
+      char upch = Ascii.toUpperCase(ch);
+      if (upch >= 'A' && upch <= 'Z' || upch == '_') {
+        name.append(upch);
+      } else if (SPECIAL_SYMBOL_MAP.containsKey(upch)) {
+        name.append(SPECIAL_SYMBOL_MAP.get(upch));
+      }
+    }
+    return name.toString();
+  }
+
+  /**
+   * Creates a key for the given string.
+   *
+   * @param s
+   *          String
+   * @return Case sensitive or case insensitive key
+   */
+  private CharSequence createKey(final String s) {
+    if (ignoreCase) {
+      return new IgnoreCaseString(s);
+    } else {
+      return s;
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */

```

Comment:
this javadoc is not needed. Can we remove it?

Reply:
ok.